internal/app/wwctl/ssh/main.go

```diff
@@ -40,21 +40,25 @@ func CobraRunE(cmd *cobra.Command, args []string) error {
 	}
 
 	for _, node := range nodes {
-
-		if _, ok := node.NetDevs["default"]; !ok {
-			fmt.Fprintf(os.Stderr, "%s: Default network device doesn't exist\n", node.Id.Get())
+		var primaryNet string
+		for netName := range node.NetDevs {
+			if node.NetDevs[netName].Primary.GetB() {
+				primaryNet = netName
```

Comment:
I suggest that there should be a `break` statement after a `Primary` interface is found. A node can have multiple interfaces marked "primary", and the current behavior uses the _last_ one; I think it should use the first one.

```bash
$ sudo cat /etc/warewulf/nodes.conf
WW_INTERNAL: 43
nodeprofiles:
  default:
    comment: This profile is automatically included for each node
nodes:
  c1:
    profiles:
    - default
    network devices:
      default:
        ipaddr: 10.0.0.100
        primary: "true"
      second:
        ipaddr: 10.0.0.101
        primary: "true"

$ sudo wwctl ssh c1 hostname # times out because this node doesn't actually exist; but shows what IP it's trying to use
c1: ssh: connect to host 10.0.0.101 port 22: Connection timed out
```

Reply:
A break would be faster, but maps are not ordered in golang, so it does not matter if I use a break or continue.
A general plausibilty check of `nodes.conf` would be needed, before saving it.